app/src/main/java/com/example/edwinmperazaduran/gridimagesearch/activities/SearchActivity.java

```diff
@@ -0,0 +1,179 @@
+import android.support.v4.app.FragmentManager;
+import android.support.v7.app.AppCompatActivity;
+import android.view.Menu;
+import android.view.MenuItem;
+import android.view.View;
+import android.view.inputmethod.InputMethodManager;
+import android.widget.AdapterView;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.GridView;
+import android.widget.Toast;
+
+import com.example.edwinmperazaduran.gridimagesearch.Dialogs.ImageFilterDialog;
+import com.example.edwinmperazaduran.gridimagesearch.R;
+import com.example.edwinmperazaduran.gridimagesearch.adapters.ImageResultArrayAdapter;
+import com.example.edwinmperazaduran.gridimagesearch.helpers.EndlessScrollListener;
+import com.example.edwinmperazaduran.gridimagesearch.models.ImageFilter;
+import com.example.edwinmperazaduran.gridimagesearch.models.ImageResult;
+import com.example.edwinmperazaduran.gridimagesearch.net.SearchClient;
+import com.loopj.android.http.JsonHttpResponseHandler;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.util.ArrayList;
+
+import cz.msebera.android.httpclient.Header;
+
+public class SearchActivity extends AppCompatActivity implements
+        ImageFilterDialog.ImageFilterDialogListener{
+
+    private static int MAX_PAGE = 8;
+    EditText etQuery;
+    GridView gvResults;
+    Button btnSearch;
+    ArrayList<ImageResult> imageResults;
+    ImageResultArrayAdapter imageAdapter;
+    SearchClient client;
+    int startPage = 0;
+    String query;
+    ImageFilter imageFilter = new ImageFilter();
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_search);
+        setupViews();
+    }
+
+    public void setupViews(){
+        etQuery = (EditText) findViewById(R.id.etQuery);
+        gvResults = (GridView) findViewById(R.id.gvResults);
+        btnSearch = (Button) findViewById(R.id.btnSearch);
+
+        btnSearch.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                hideSoftKeyboard(v);
+                onImageSearch(0);
+            }
+        });
+
+        gvResults.setOnItemClickListener(new AdapterView.OnItemClickListener() {
+            @Override
+            public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
+                Intent i = new Intent(getApplicationContext(), ImageDisplayActivity.class);
+                ImageResult imageResult = imageResults.get(position);
+                i.putExtra("result", imageResult);
+                startActivity(i);
+            }
+        });
+
+        gvResults.setOnScrollListener(new EndlessScrollListener() {
+            @Override
+            public void onLoadMore(int page, int totalItemsCount) {
+                if (page < MAX_PAGE) {
+                    onImageSearch(page - 1);
```

Comment:
@edwinperaza -1 logic should be handled inside the `onImageSearch()` method. This way you just pass in the page and thats it.